core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -36,6 +40,42 @@
  */
 public interface ClientRequestContext extends RequestContext {
 
+    /**
+     * Returns a new {@link ClientRequestContext} created from the specified {@link HttpRequest}.
+     * Note that it is not usually required to create a new context by yourself, because Armeria
```

Comment:
Should we go ahead and call it `FakeClientRequestContextBuilder` and put it in the `testing` package? I feel like this really does have no real point outside of unit tests.

Reply:
Ah oops, I missed the second point of `motivation` - is there an example of Sometimes a user want to emulate an incoming request and feed it into his or her processing pipeline. ? Wondering what it means.

Reply:
Updated the PR description with an example in the motivation section. :-)